libuavcan/include/libuavcan/platform/memory.hpp

```diff
@@ -304,6 +323,126 @@ class LIBUAVCAN_EXPORT PoolAllocator
     }
 };
 
+namespace
+{
+/*
+ * Internal template used by copyBits. Don't use this. Use copyBits.
+ * This method does not check the input pointers for null nor does it
+ * validate its template parameters.
+ *
+ * @tparam SrcType          The unsigned integer type of the source memory.
+ * @tparam DstType          The unsigned integer type of the destination memory.
+ * @param  src              The memory location to read from.
+ * @param  src_bit_offset   The the number of bits into @p src to start reading
+ *                          from.
+ * @param  dst              The memory location to write to.
+ * @param  dst_bit_offset   The number of bits into @p dst to start writing to.
+ * @param  write_mask       A mask to apply to the shifted @p src before writing
+ *                          to the shifted @p dst.
+ */
+template <typename SrcType, typename DstType>
+void _unalignedUnsafeWrite(const SrcType* src,
+                           std::size_t    src_bit_offset,
+                           DstType*       dst,
+                           std::size_t    dst_bit_offset,
+                           const DstType  write_mask)
+{
+    *dst = DstType((*dst & ~write_mask) | (DstType((*src >> src_bit_offset) << dst_bit_offset) & write_mask));
+}
+}  // namespace
+
+/**
+ * This function implements a portable copy of unaligned bit arrays that avoids generation of
+ * unaligned instructions for a platform.
+ *
+ * Note that this implementation makes a strong assumption that the platform is
+ * little endian. This logic should completely fail on a big-endian system.
```

Comment:
Shouldn't this be guarded by a preprocessor check? If we could detect C++20 it could be a constexpr check https://en.cppreference.com/w/cpp/types/endian

People use libuavcan on big-endian machines, see #205, #210

Reply:
added #267 but will also make this work on 8-bit words only. I assume we don't plan on supporting 4-bit processors? ;-)

Reply:
Hail standardization.